### PR TITLE
Replace purrr functions with base/BBmisc

### LINKED ISSUE #2489: What to do with the purrr pkg?
Bernd raised the question on Slack.
https://cran.r-project.org/web/packages/purrr/index.html
_purrr_ deps:

Imports: | magrittr (≥ 1.5), rlang (≥ 0.1), tibble
-- | --
Suggests: | covr, dplyr (≥ 0.7.5), knitr, rmarkdown, testthat

- Have _purrr_ staying in IMPORTS (which it is currently since some months)
- Move it to SUGGESTS
- A complete removal is not really possible since we rely on the `imap()` function in some code parts which is not available elsewhere.

### PR DESCRIPTION
**Pull request by pat-s**
fixes #2489

### CODE REVIEW

**Comment by larskotthoff** [comment]
This seems to break a few things...

File: R/createSpatialResamplingPlots.R [hunk 1]
```diff
@@ -180,11 +180,12 @@ createSpatialResamplingPlots = function(task = NULL, resample = NULL, crs = NULL
       theme(axis.text.x = element_text(size = axis.text.size),
         axis.text.y = element_text(size = axis.text.size),
         plot.margin = unit(c(0.5, 0.2, 0.2, 0.2), "cm"))
+      }
     )
     return(plot.list.out)
   })
 
-  plot.list = flatten(plot.list.out.all)
+  plot.list = unlist(plot.list.out.all)
```
Comment (mb706) [inline R/createSpatialResamplingPlots.R]: probably `recursive = FALSE`

File: R/getResamplingIndices.R [hunk 2]
```diff
@@ -45,17 +44,17 @@ getResamplingIndices = function(object, inner = FALSE) {
...
+        lapply(c("train.inds", "test.inds"), function(u) # map over train/test level
 
           # list() -> create list for "train.inds" and "test.inds"
-          # flatten() -> reduce by one level
+          # unlist() -> reduce by one level
           # set_names(c("train.inds", "test.inds")) -> now set list names
-          flatten(
-            map(inner_inds[[z]][[u]], ~ # map over number of inner folds
-                  list(outer_inds[["train.inds"]][[z]][.x]) # the inner test.inds are a subset of the outer train.inds! That's why "train.inds" is hardcoded here
+          unlist(
```
Comment (mb706) [inline R/getResamplingIndices.R]: probably should use `recursive = FALSE`, but as I say below I dont really understand what you are trying to do here.

**Comment by pat-s** [comment]
@mb706 	Thanks for taking the time to review this. Would have been a nice review before we merged it :)

I am sometimes unsure if all your reviews are on the positive or negative side in relation to me - they sound somewhat critical from time to time ;)

I took the time and addressed all your points and learned sth - note that I am not an IT guy by education but self-trained, i.e. such kind of things can possibly happen more often.

**Comment by mb706** [comment]
I agree that it would be nice to have tighter code reviews, but I'm also aware of the obvious costs of time and work, and as long as the tests pass and stuff works this will probably outweigh the limited short term benefits from consistent style and idiomatic code.

Once the linter is working again it should be helpful in this regard however.

**Comment by larskotthoff** [comment]
Thanks, merging.

**Pull request merged**